bin/plugin/commands/test-plugins.js

```diff
@@ -458,23 +460,67 @@ function doRunStandalonePluginTests( settings ) {
 	}
 
 	// Create an array of plugins from entries in plugins JSON file.
-	builtPlugins = Object.keys( plugins )
+	builtPlugins = Object.keys( stPlugins )
 		.filter( ( item ) => {
 			if (
 				! fs.pathExistsSync(
-					`${ settings.builtPluginsDir }${ plugins[ item ].slug }`
+					`${ settings.builtPluginsDir }${ stPlugins[ item ].slug }`
 				)
 			) {
 				log(
 					formats.error(
-						`Built plugin path "${ settings.builtPluginsDir }${ plugins[ item ].slug }" not found, skipping and removing from plugin list`
+						`Built plugin path "${ settings.builtPluginsDir }${ stPlugins[ item ].slug }" not found, skipping and removing from plugin list`
 					)
 				);
 				return false;
 			}
 			return true;
 		} )
-		.map( ( item ) => plugins[ item ].slug );
+		.map( ( item ) => stPlugins[ item ].slug );
+
+	// Append plugins into array.
+	const plugins = pluginsConfig.plugins;
+	if ( ! plugins ) {
+		log(
+			formats.error(
+				'The given plugin configuration is invalid, the plugins are missing, or they are misspelled.'
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Create an array of plugins from entries in plugins JSON file.
+	builtPlugins = builtPlugins.concat(
+		Object.keys( plugins )
+			.filter( ( item ) => {
+				try {
+					fs.copySync(
+						`${ settings.pluginsDir }${ plugins[ item ].slug }/`,
+						`${ settings.builtPluginsDir }${ plugins[ item ].slug }/`,
+						{
+							overwrite: true,
+						}
+					);
+					log(
+						formats.success(
+							`Copied plugin "${ plugins[ item ].slug }".\n`
+						)
+					);
+					return true;
+				} catch ( e ) {
+					// Handle the error appropriately
+					log(
+						formats.error(
+							`Error copying plugin "${ plugins[ item ].slug }": ${ e.message }`
+						)
+					);
+					return false;
+				}
+			} )
+			.map( ( item ) => plugins[ item ].slug )
```

Comment:
As elsewhere, this can be simplified since the object keys are not used:

```suggestion
		Object.values( plugins )
			.filter( ( plugin ) => {
				try {
					fs.copySync(
						`${ settings.pluginsDir }${ plugin.slug }/`,
						`${ settings.builtPluginsDir }${ plugin.slug }/`,
						{
							overwrite: true,
						}
					);
					log(
						formats.success(
							`Copied plugin "${ plugin.slug }".\n`
						)
					);
					return true;
				} catch ( e ) {
					// Handle the error appropriately
					log(
						formats.error(
							`Error copying plugin "${ plugin.slug }": ${ e.message }`
						)
					);
					return false;
				}
			} )
			.map( ( plugin ) => plugin.slug )
```